File: python/Euler_012.py
def compute():
    num = 0
    i = 1
    while True:
        num += i
        divisors = sum(2 for j in range(1, int(num ** 0.5) + 1) if num % j == 0)
        if num ** 0.5 % 1 == 0:
            divisors -= 1
        if divisors > 500:
            return num
        i += 1


print(compute())

File: java/Euler_012.java

public class Euler_012 {

    public static void main(String[] args) {
        int num = 0;

        for (int i = 1;; i++) {
            int divisors = 0;
            num += i;

            for (int j = 1; j <= Math.sqrt(num); j++) {
                if (num % j == 0) {
                    divisors += 2;
                }
                if (Math.sqrt(num) % 1 == 0) {
                    divisors--;
                }
            }

            if (divisors > 500) {
                System.out.println(num);
                return;
            }
        }
    }

}
